lib/spack/spack/cmd/buildcache.py

```diff
@@ -767,6 +779,74 @@ def buildcache_copy(args):
         shutil.copyfile(cdashid_src_path, cdashid_dest_path)
 
 
+def buildcache_sync(args):
+    """ Syncs binaries (and associated metadata) from one mirror to another.
+    Requires an active environment in order to know which specs to sync.
+
+    Args:
+        src (str): Source mirror URL
+        dest (str): Destination mirror URL
+    """
+    if not args.src_mirror or not args.dest_mirror:
+        tty.die('Source and destination mirror urls are required')
+
+    src_mirror_url = args.src_mirror
+    dest_mirror_url = args.dest_mirror
```

Comment:
This should also accept the names of pre-configured mirrors.  See `_createtarball` for an example of how this is done.